packages/fds-components/src/components/interactive/Chips.jsx

```diff
@@ -0,0 +1,111 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import cx from 'classnames';
+
+import DenyIcon from 'fds-icons/lib/react/DenyIcon';
+
+/**
+ * Common pattern where we use href to generate whether
+ * an href or a button is returned, and also allow a master
+ * override where you can specify any element / component as
+ * the root
+ */
+const baseElement = ({ href, rootEl }) => {
+  let Element;
+  if (rootEl) {
+    Element = rootEl;
+  } else if (href) {
+    Element = 'a';
+  } else {
+    Element = 'button';
+  }
+  return Element;
+};
+
+const Chip = ({
+  size,
+  rootEl,
+  theme,
+  isActive,
+  children,
+  label,
+  subtitle,
+  hasClose,
+  ...rest
+}) => {
+  const Element = baseElement({ href: rest.href, rootEl });
+
+  const rootClass = cx('fdsChip', {
+    'fdsChip--active': isActive,
+    'fdsChip--small': size === 'small',
+    'fdsChip--large': size === 'large',
+    'fdsChip--blue': theme === 'blue',
+    'fdsChip--outline': theme === 'outline',
+  });
+  return (
+    <Element {...rest} className={rootClass}>
+      {label}
+      {subtitle && <span className="fdsChip-subtitle">{subtitle}</span>}
+      {hasClose && (
+        <span className="fdsChip-close">
+          <DenyIcon size="xs" />
+        </span>
+      )}
+    </Element>
+  );
+};
+
+Chip.propTypes = {
+  label: PropTypes.string,
+  /**
+   * This allows for overriding of the Chip root element.
+   * Meant to accomodate  `<Link />`
+   */
+  rootEl: PropTypes.func,
+  isActive: PropTypes.bool,
+  subtitle: PropTypes.string,
+  className: PropTypes.string,
+  children: PropTypes.any,
+  hasClose: PropTypes.bool,
+  value: PropTypes.string,
+  size: PropTypes.oneOf(['small', 'medium', 'large']),
+  theme: PropTypes.oneOf(['blue', 'gray', 'outline']),
+};
+
+const Chips = ({ list, ...topRest }) => (
+  <ul className="fdsChip-ul">
+    {list.map((listItem) => (
+      <li className="fdsChip-li" key={listItem.value || listItem.label}>
```

Comment:
I think it's possible to have the same value/label here. Combine the value/label with the `index` could help keep the key unique.

Reply:
Thanks for reviewing!

So the only problem with that, is that keys aren't supposed to be based off of the index. Essentially the idea behind keys, is let's say you have 10 items. If you remove the 3rd item, you want to communicate to react to remove only that 3rd item. If you have the keys based on index, and you remove the 3rd item, the index of every item after that will update.

https://medium.com/@robinpokorny/index-as-a-key-is-an-anti-pattern-e0349aece318

Reply:
Yes, use index is no good for performance. But if we have the same key, the list will break.

Reply:
Totally. So for the most part, it's the consumer's responsibility to make sure to provide unique keys, not only for their own components, but for this one too.

The great thing about us controlling the mapping, is that consumers are unable to mistakenly use index as a key, so they're forced to find a better way to establish a key.

For this component, by default, it attempts to use the label as the key. Ostensibly, labels (and by proxy tags and filters) in the application "really" shouldn't be duplicated in the application (correct me if I'm wrong though)... so this should provide a correct key in 99% of cases without interaction (and protects against using index)

For the 1% case, the consumer will need to provide their own key (maybe a unique id they get from the backend... maybe they decide to generate their own uuid, etc). But yeah that's totally expected, as there's no real practical way to enforce the id's are unique on the reusable component side. The developer would see the duplicate key error in their console, and then look at the component interface to figure out how to pass a unique identifier. That make any sense at all?

The only bad thing about all of this: React actually will throw an error if you try to create a "key" prop. The name "value" could potentially be a bit more distinctive and establish a pattern across fds components?

Reply:
I agree. Many thanks for the explanation!

Reply:
absolutely, thanks for the review.